jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -370,33 +376,33 @@ def vue_do_aper_phot(self, *args, **kwargs):
                                               label=comp.units or 'Value')]
 
                 if self.current_plot_type == "Radial Profile":
-                    self._fig.title = 'Radial profile from source centroid'
+                    self._fig.title = 'Radial profile from aperture center'
                     x_data, y_data = _radial_profile(
-                        phot_aperstats.data_cutout, phot_aperstats.bbox, phot_aperstats.centroid,
+                        phot_aperstats.data_cutout, phot_aperstats.bbox, (xcenter, ycenter),
                         raw=False)
                     bqplot_line = bqplot.Lines(x=x_data, y=y_data, marker='circle',
                                                scales={'x': line_x_sc, 'y': line_y_sc},
                                                marker_size=32, colors='gray')
                 else:  # Radial Profile (Raw)
-                    self._fig.title = 'Raw radial profile from source centroid'
+                    self._fig.title = 'Raw radial profile from aperture center'
                     x_data, y_data = _radial_profile(
-                        phot_aperstats.data_cutout, phot_aperstats.bbox, phot_aperstats.centroid,
+                        phot_aperstats.data_cutout, phot_aperstats.bbox, (xcenter, ycenter),
                         raw=True)
                     bqplot_line = bqplot.Scatter(x=x_data, y=y_data, marker='circle',
                                                  scales={'x': line_x_sc, 'y': line_y_sc},
                                                  default_size=1, colors='gray')
 
                 # Fit Gaussian1D to radial profile data.
-                # mean is fixed at 0 because we recentered to centroid.
                 if self.fit_radial_profile:
                     fitter = LevMarLSQFitter()
                     y_max = y_data.max()
+                    x_mean = x_data[np.where(y_data == y_max)].mean()
                     std = 0.5 * (phot_table['semimajor_sigma'][0] +
                                  phot_table['semiminor_sigma'][0])
                     if isinstance(std, u.Quantity):
                         std = std.value
-                    gs = Gaussian1D(amplitude=y_max, mean=0, stddev=std,
-                                    fixed={'mean': True, 'amplitude': True},
+                    gs = Gaussian1D(amplitude=y_max, mean=x_mean, stddev=std,
+                                    fixed={'amplitude': True},

```

Comment:
It still isn't fool-proof (it will fail if `y_max` returns `x_data` values at different peaks, for instance) but hopefully work good enough for most cases that this plugin is useful for... 🤞